docs/en/sql-reference/aggregate-functions/reference/timeSeriesDeltaToGrid.md

```diff
@@ -0,0 +1,69 @@
+---
+description: 'Aggregate function that calculates PromQL-like delta over time series data on the specified grid.'
+sidebar_position: 221
+slug: /sql-reference/aggregate-functions/reference/timeSeriesDeltaToGrid
+title: 'timeSeriesDeltaToGrid'
+---
+
+Aggregate function that takes time series data as pairs of timestamps and values and calculates [PromQL-like delta](https://prometheus.io/docs/prometheus/latest/querying/functions/#delta) from this data on a regular time grid described by start timestamp, end timestamp and step. For each point on the grid the samples for calculating `delta` are considered within the specified time window.
+
+Parameters:
+- `start timestamp` - specifies start of the grid
+- `end timestamp` - specifies end of the grid
+- `grid step` - specifies step of the grid in seconds
+- `window` - specified the maximum "staleness" in seconds of the considered samples
+
+Arguments:
+- `timestamp` - timestamp of the sample
+- `value` - value of the time series corresponding to the `timestamp`
+
+Return value:
+`delta` values on the specified grid as an `Array(Nullable(Float64))`. The returned array contains one value for each time grid point. The value is NULL if there is no sample for a particular grid point.
```

Comment:
Actually `delta` requires two points to calculate (first and last). So if there are less than two points in interval then the result should NULL.

Reply:
Modified the docs